test/testcafe/spec/DeviceChallengePollViewFailure_spec.js

```diff
@@ -10,8 +11,11 @@ const mock = RequestMock()
   .respond(identifyWithDeviceProbingLoopback)
   .onRequestTo(/\/idp\/idx\/authenticators\/poll/)
   .respond((req, res) => {
-    res.statusCode = '403';
-    res.setBody(error);
+    return new Promise((resolve) => setTimeout(function() {
+      res.statusCode = '403';
+      res.setBody(error);
+      resolve(res);
+    }, Constants.TESTCAFE_DEFAULT_AJAX_WAIT + 2000));
```

Comment:
why does this response need to be delayed?

Reply:
Probably to mimic the poll

Reply:
We should not have such delays in our tests. Could this be done better with a unit test that mocks time?

Reply:
While i also had concern regarding these delays.. but we need to test login flows and couldn't find any solutions yet for testcafe to mimic these delays.. Unit test won't be sufficient.